staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -322,6 +324,42 @@ func (b *builder) buildKubeNative(schema *structuralschema.Structural, v2 bool)
 	return ret
 }
 
+func addEmbeddedProperties(s *spec.Schema) {
```

Comment:
ideally I'd expect this to happen in [unfold](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/unfold.go) or at least before `ToStructuralOpenAPIV2` conversion. I can see it being here for now given the builtin schemata only use v2 features, and depend on go-openapi heavily

Reply:
true. That's why we have unfold.

Reply:
I initially added it there but it made round tripping weird